packages/hash/api/src/model/ontology/dataType.model.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  DataType as DataTypeSchema,
+  GraphApi,
+} from "@hashintel/hash-graph-client";
+
+import { DataType as DataTypeModel } from "../index";
+
+type DataTypeArgs = {
+  accountId: string;
+  schema: DataTypeSchema;
+};
+
+class __DataType {
+  accountId: string;
+
+  schema: DataTypeSchema;
+
+  constructor({ schema, accountId }: DataTypeArgs) {
+    this.accountId = accountId;
+    this.schema = schema;
+  }
+
+  /**
+   * Create a data type.
+   *
+   * User defined data types are not specified yet, which means this `create`
+   * operations should not be exposed to users yet.
+   *
+   * @param params.accountId the accountId of the account creating the data type
+   * @param params.schema a `DataType`
+   */
+  static async create(
+    graphApi: GraphApi,
+    params: {
+      accountId: string;
+      schema: DataTypeSchema;
+    },
+  ): Promise<DataTypeModel> {
+    const { data: schema } = await graphApi.createDataType(params);
+
+    return new DataTypeModel({ schema, accountId: params.accountId });
+  }
+
+  /**
+   * Get latest versions of all data types.
```

Comment:
I'd probably rephrase as
```suggestion
   * Get all data types at their latest version.
```